### PR TITLE
PLT-8693: Add Node.js support

### PR DESCRIPTION
**Pull request by hrajchert**
This PR adds the ability to use the SDK from node.js (and deno) by adding a new wallet implementation on top of the Lucid library and by allowing a new way to create a `runtime-lifecycle` object from a Wallet implementation.

An example of usage for Node.js was included, a Deno example will follow once this is published on NPM, but it should work similarly. 

To test this PR follow the new instructions in `examples/nodejs/Readme.md`

## Summary by CodeRabbit

- **New Features**
  - Introduced Node.js support for the TS-SDK with setup instructions and examples.
  - Added a search bar to the app interface for enhanced user experience.
  - Implemented a new escrow-flow script for asset escrowing on the Cardano blockchain.
  - Created a wallet-flow script for performing various wallet operations.
  - Introduced a Lucid implementation for the wallet API compatible with Browser, Node.js, and Deno environments.
  - Added a top-level function `mkRuntimeLifecycle` that accepts a wallet implementation.

- **Documentation**
  - Updated the Readme.md with new example usage and setup instructions for Node.js projects.
  - Provided additional contextual annotations (`@experimental` and `@remarks`) for methods in the `WalletAPI` interface.

- **Bug Fixes**
  - Addressed unspecified bugs in the asset management logic.

- **Refactor**
  - Refactored the `mkRuntimeLifecycle` function to take `runtimeURL` and `wallet` as arguments.
  - Updated import statements and type references in the wallet package to align with the new Lucid module structure.

- **Chores**
  - Added `.config.json` to the .gitignore file for Node.js examples.
  - Modified CI/CD workflow environment variables for Nix and Cachix setup.

- **Style**
  - Enhanced styles for the new search bar feature.

- **Tests**
  - Renamed a job in the test suite workflow to reflect its purpose accurately.

### CODE REVIEW

File: .github/workflows/test-suite.yaml [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+  SUBSTITUTERS: "https://cache.nixos.org/ https://cache.iog.io https://marlowe-temp.cachix.org https://cache.zw3rk.com/"
 on:
   pull_request:
     types:
       - opened
       - synchronize
 
 jobs:
-  check-changelog:
+  test-suite:
```
Comment (nhenin) [inline .github/workflows/test-suite.yaml:14]: I just momentarily forgot we didn't completely disable the tests but only the integration ones...

**Food for thoughts** : We should have a document in the repo, explaining what is the current level testing-wise and some directions how we want to improve it...

File: examples/nodejs/Readme.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Using this example
+
+To use this example you first need to install dependencies and build the code
```
Comment (nhenin) [inline examples/nodejs/Readme.md:3]: I don't know if it is a French thing, but I usually avoid saying "you" when explaining something...

Comment (nhenin) [reply examples/nodejs/Readme.md:3]: e.g : To use this example, install the dependencies and build the codebase

File: examples/nodejs/src/escrow-flow.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import arg from "arg";
```
Comment (nhenin) [inline examples/nodejs/src/escrow-flow.ts:1]: I like this example... It should be well visible, maybe a FAQ ?  Like `I want to use Marlowe with NodeJS or/and Deno ? How can I do it ?`

File: examples/nodejs/src/wallet-flow.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { mkLucidWallet } from "@marlowe.io/wallet";
```
Comment (nhenin) [inline examples/nodejs/src/wallet-flow.ts:1]: Cool !

File: packages/wallet/src/api.ts [hunk 5]
```diff
@@ -97,6 +123,13 @@ export interface WalletAPI {
...
    * A helper around `getTokens` that returns only the lovelace ammount.
    * @experimental
+   * This is marked experimental as it is redundant to have both getTokens and getLovelaces for the implementors.
+   * We could offer this as a function on top of getTokens, the only reason to have it here is for possible
+   * performance improvements.
+   *
+   * Moreover, this function is useful in the context of DApps to show the user how much lovelaces they have
+   * but not for the context of creating or balancing Marlowe transactions.
+   * @see {@link https://github.com/input-output-hk/marlowe-ts-sdk/discussions/106 | Github discussion on wallet package scope}
    */
```
Comment (nhenin) [inline packages/wallet/src/api.ts:133]: Great

**Review by nhenin — APPROVED** [review]

File: examples/nodejs/src/escrow-flow.ts (hunk 3 above)
Comment (hrajchert) [reply examples/nodejs/src/escrow-flow.ts:1]: eventually yeah! that is why I added @nstanford5 to review this as well

**Review by nstanford5 — APPROVED** [review]

File: examples/nodejs/Readme.md (hunk 2 above)
Comment (bjornkihlberg) [reply examples/nodejs/Readme.md:3]: Not using personal pronouns ("you", "we","I", "they", etc) makes text more formal. I don't have strong feelings on this matter but I tend to not use personal pronouns when recounting events in documentation but I think when providing instructions like in this case, using "you" is not as bad.

**Pull request merged**